core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsParser.java

```diff
@@ -19,89 +19,59 @@
-            } else if (context.matchField(currentFieldName, SignificantTermsAggregationBuilder.BACKGROUND_FILTER)) {
-                QueryParseContext queryParseContext = new QueryParseContext(context.getDefaultScriptLanguage(), queriesRegistry,
-                        context.getParser(), context.getParseFieldMatcher());
-                Optional<QueryBuilder> filter = queryParseContext.parseInnerQueryBuilder();
-                if (filter.isPresent()) {
-                    otherOptions.put(SignificantTermsAggregationBuilder.BACKGROUND_FILTER, filter.get());
-                }
-                return true;
-            }
+        parser.declareInt(SignificantTermsAggregationBuilder::shardSize, TermsParser.SHARD_SIZE_FIELD_NAME);
+
+        parser.declareLong(SignificantTermsAggregationBuilder::minDocCount, TermsParser.MIN_DOC_COUNT_FIELD_NAME);
+
+        parser.declareLong(SignificantTermsAggregationBuilder::shardMinDocCount, TermsParser.SHARD_MIN_DOC_COUNT_FIELD_NAME);
+
+        parser.declareInt(SignificantTermsAggregationBuilder::size, TermsParser.REQUIRED_SIZE_FIELD_NAME);
+
+        parser.declareString(SignificantTermsAggregationBuilder::executionHint, TermsParser.EXECUTION_HINT_FIELD_NAME);
+
+        parser.declareObject((b, v) -> { if (v.isPresent()) b.backgroundFilter(v.get()); },
+                (parser, context) -> context.parseInnerQueryBuilder(),
+                SignificantTermsAggregationBuilder.BACKGROUND_FILTER);
+
+        parser.declareField((b, v) -> b.includeExclude(IncludeExclude.merge(v, b.includeExclude())),
+                IncludeExclude::parseInclude, IncludeExclude.INCLUDE_FIELD, ObjectParser.ValueType.OBJECT_ARRAY_OR_STRING);
+
+        parser.declareField((b, v) -> b.includeExclude(IncludeExclude.merge(b.includeExclude(), v)),
+                IncludeExclude::parseExclude, IncludeExclude.EXCLUDE_FIELD, ObjectParser.ValueType.STRING_ARRAY);
+
+        for (String name : significanceHeuristicParserRegistry.getNames()) {
+            parser.declareObject(SignificantTermsAggregationBuilder::significanceHeuristic,
+                    (parser, context) -> {
+                        SignificanceHeuristicParser significanceHeuristicParser = significanceHeuristicParserRegistry
+                                .lookupReturningNullIfNotFound(name, context.getParseFieldMatcher());
+                        return significanceHeuristicParser.parse(context);
+                    },
+                    new ParseField(name));
         }
-        return false;
     }
 
     @Override
-    protected BucketCountThresholds getDefaultBucketCountThresholds() {
-        return new TermsAggregator.BucketCountThresholds(SignificantTermsAggregationBuilder.DEFAULT_BUCKET_COUNT_THRESHOLDS);
+    public AggregationBuilder parse(String aggregationName, QueryParseContext context) throws IOException {
+        return parser.parse(context.parser(), new SignificantTermsAggregationBuilder(aggregationName, null), context);
     }
+
```

Comment:
OK - I've made it this far and I'm going to stop here for this round. I think I've left enough feedback for a lively discussion!